src/ocr/ocr.cpp

```diff
@@ -130,11 +132,23 @@ QStringList Ocr::availableLanguages(const QString &languagesPath)
 
 void Ocr::setParameters(const QMap<QString, QVariant> &parameters, bool saveSettings)
 {
+    // Restore changed values before to defaults
+    for (auto it = m_defaultParameters.cbegin(); it != m_defaultParameters.cend(); ++it)
+        m_tesseract.SetVariable(it.key(), it.value());
+    m_defaultParameters.clear();
+
+    // Apply new parameters
     for (auto it = parameters.cbegin(); it != parameters.cend(); ++it) {
-        if (!m_tesseract.SetVariable(it.key().toLocal8Bit(), it.value().toByteArray()))
+        const QByteArray key = it.key().toLocal8Bit();
+        const QByteArray value = it.value().toByteArray();
+        const QByteArray defaultValue = m_tesseract.GetStringVariable(key);
```

Comment:
Unfortunately, this line seems to segfault unless we previously set the variable value ourselves.

Reply:
Strange, I thought `GetStringVariable` would return the default variable if it wasn't set before.

We already have a reinit on languages change (`setLanguagesString`), so I need to move this to that function.

Reply:
Also, `m_tesseract.GetStringVariable()` only works for variables that take a string value, so we would want to use `m_tesseract.GetVariableAsString()` which works for booleans and numbers as well but converts the result to a string. Either way, it also doesn't work unless we've set the value ourselves.